src/storage/temp.rs

```diff
@@ -7,23 +7,25 @@ use std::path::PathBuf;
 
 use tempfile::NamedTempFile;
 
-use super::{StorageProvider, StorageReader};
+use super::StorageProvider;
 use crate::WrapIoResult;
 
-/// Creates a [TempStorageReader] backed by a temporary file
+/// Creates a [`TempStorageReader`] backed by a temporary file
 #[derive(Default, Clone, Debug)]
 pub struct TempStorageProvider {
     storage_dir: Option<PathBuf>,
 }
 
 impl TempStorageProvider {
-    /// Creates a new [TempStorageProvider] that creates temporary files in the OS-specific default
+    /// Creates a new [`TempStorageProvider`] that creates temporary files in the OS-specific default
     /// location.
     pub fn new() -> Self {
-        Self { storage_dir: None }
+        Self {
+            storage_dir: None,
```

Comment:
Looks like one more rustfmt error here

Reply:
might be an idea to setup a github action to auto format.

Reply:
Is that a common practice? Don't think I've seen that before. I'm not a huge fan of bots auto-committing directly into active branches and I figure most people already use rustfmt anyway.

Reply:
I have seen it here and there, usually the bot changes the PR not the main branch. It can be nice as the auto formatting is done by a neutral party so it does not skew contribution numbers. And it can make reviewing PR's slightly easier.  I dont see it being used on *rust-analyzer* or *tokio* however (to me examples of a great rust codebases). 

A middle ground, you can use `rustfmt -- --check` to simply fail CI if the formatting is not up to spec. Thats what the projects above do.

Oh and for me GitHub CI is an absolute pain in the *** to setup, so don't waste your time on it.

Reply:
I do have a rustfmt check that fails the CI, that's where I saw the formatting error: https://github.com/aschey/stream-download-rs/actions/runs/6215931544/job/16869320544#step:7:53